File: 0374. Guess Number Higher or Lower.py
# O(log n) time and O(1) space complexity
class Solution:
    def guessNumber(self, n: int) -> int:
        num = int(n/2)
        Min, Max = 0, n

        while True:
            g = guess(num)

            # guess is correct
            if g == 0:
                return num

            # guess is too low
            elif g == 1:
                Min = num
                num = int((Min + Max) / 2) + 1

            # guess is too high
            elif g == -1:
                Max = num
                num = int((Min + Max) / 2)

File: 0374. Guess Number Higher or Lower.js
// O(log n) time and O(1) space complexity
var guessNumber = function(n) {
    let num = Number(n / 2);
    let min = 0, max = n;

    while(true) {
        let g = guess(num);

        // guess is correct
        if (g == 0) { return num }

        // guess is too low
        else if (g == 1) {
            min = num;
            num = Number((min + max) / 2) + 1;
        }

        // guess is too high
        else if (g == -1) {
            max = num;
            num = Number((min + max) / 2);
        };
    };
};
